jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ContextProvider.java

```diff
@@ -131,161 +136,186 @@ public ContextProvider()
-        setExtract(extract);
-    }
-
-    /**
-     * Set to extract WAR and JAR files.
-     *
-     * @param extract the extractWars to set
-     */
-    public void setExtract(boolean extract)
-    {
-        _extract = extract;
+        _properties.put("jetty.deploy.extractWars", Boolean.toString(extractWars));
     }
 
     /**
      * Get the parentLoaderPriority.
+     * This is equivalent to the <code>jetty.deploy.parentLoaderPriority</code> property.
      *
      * @return the parentLoaderPriority
      */
     @ManagedAttribute("parent classloader has priority")
     public boolean isParentLoaderPriority()
     {
-        return _parentLoaderPriority;
+        return Boolean.parseBoolean(_properties.get("jetty.deploy.parentLoaderPriority"));
     }
 
     /**
      * Set the parentLoaderPriority.
+     * This is equivalent to the <code>jetty.deploy.parentLoaderPriority</code> property.
      *
      * @param parentLoaderPriority the parentLoaderPriority to set
      */
     public void setParentLoaderPriority(boolean parentLoaderPriority)
     {
-        _parentLoaderPriority = parentLoaderPriority;
+        _properties.put("jetty.deploy.parentLoaderPriority", Boolean.toString(parentLoaderPriority));
     }
 
     /**
      * Get the defaultsDescriptor.
+     * This is equivalent to the <code>jetty.deploy.defaultsDescriptor</code> property.
      *
      * @return the defaultsDescriptor
      */
     @ManagedAttribute("default descriptor for webapps")
     public String getDefaultsDescriptor()
     {
-        return _defaultsDescriptor;
+        return _properties.get("jetty.deploy.defaultsDescriptor");
     }
 
     /**
      * Set the defaultsDescriptor.
+     * This is equivalent to the <code>jetty.deploy.defaultsDescriptor</code> property.
      *
      * @param defaultsDescriptor the defaultsDescriptor to set
      */
     public void setDefaultsDescriptor(String defaultsDescriptor)
     {
-        _defaultsDescriptor = defaultsDescriptor;
-    }
-
-    public ConfigurationManager getConfigurationManager()
-    {
-        return _configurationManager;
+        _properties.put("jetty.deploy.defaultsDescriptor", defaultsDescriptor);
     }
 
     /**
-     * Set the configurationManager.
-     *
-     * @param configurationManager the configurationManager to set
+     * This is equivalent to the <code>jetty.deploy.configurationClasses</code> property.
+     * @param configurations The configuration class names as a comma separated list
      */
-    public void setConfigurationManager(ConfigurationManager configurationManager)
+    public void setConfigurationClasses(String configurations)
     {
-        updateBean(_configurationManager, configurationManager);
-        _configurationManager = configurationManager;
+        setConfigurationClasses(StringUtil.isBlank(configurations) ? null : configurations.split(","));
     }
 
     /**
+     * This is equivalent to the <code>jetty.deploy.configurationClasses</code> property.
```

Comment:
Confusing? Is this overriding the `jetty.deploy.configurationClasses` property or setting it?